### DISCUSSION TITLE
zwave-js server in Windows

### DISCUSSION THREAD

**Discussion by john55021** [zwave-js/zwave-js#5235] [General]
How does one install zwave-js server on a Windows machine. I've successfully installed node.js and nqm on the system, and when i run npm i @zwave-js/server there's lots of stuff installed. I was assuming everything the server depended on, include zwave-js would be installed, but it doesn't appear to be the actual code, mode meta data type stuff. Has anyone installed server on a Windows machine, what were the steps and what configuration files (and where are they) need to be changed, so that at a minimum I can specify my USB stick and maybe the server IP address and have it usable. Thanks in advance.

**Comment by kpine** [comment]
Use [Z-Wave JS UI](https://github.com/zwave-js/zwave-js-ui) instead, which is a fully supported application and provides the websocket server. zwave-js-server is targeted as a library/server for developers.

**Reply by john55021** [reply]
Thank you for the answer. The best I can figure out is to download the zip file and unzip it. Assuming this runs under node.js, don't I need to run some kind of npm file to so node.js knows all the dependencies, etc. Basically, at this point I'm lost. How do I run this thing? With no being a Linux person, it's a bit confusing. Thanks.

**Reply by kpine** [reply]
The zip file contains an application for Windows. You just run it like any other command line application. No Linux involved.

https://zwave-js.github.io/zwave-js-ui/#/getting-started/other-methods?id=nodejs-or-pkg-version

**Reply by john55021** [reply]
Yes, i believe i downloaded the wrong file. I'll be giving it a go on a few hours.  Thanks so much for your quick response.

**Reply by john55021** [reply]
Thanks kpine, I have everything up and running - all in pure Windows, no VM's. Using HA portable, Mosquitto, Omnilink Bridge and zwave-js-ui. Have to use a program called nssm get to HA portable and zwave running as a service. But fast as lightening and I have no idea what I'm missing out by not running on Linux - Supervisor mode?

**Reply by kpine** [reply]
Sounds fine. Just be aware Windows is not a supported platform for HA. As long as the install is minimal, it might be OK.
